### PR TITLE
feat: modularise ci workflows 

### PR DESCRIPTION
**Pull request by CjS77**
To make CI workflows more manageable and allow us to run some cheaper
tests on non-core development (e.g. launchpad), this PR does the
following:

- breaks out jobs into reuasable workflows
- adds several parameters to these workflows to DRY up the CI

- adds separate workflows for specific application branches while
  keeping the core CI flow the same.

- renames workflow files to be more indicative of when they run & what
they do.

- Delete some redundant workflows

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -16,217 +16,51 @@ on:
...
-        with:
-          command: check
-          args: --release --package tari_wallet_ffi
+    uses:
+      ./.github/workflows/reusable-ci-jobs.yml
+    with:
+      check-ffi: true
+
+  check-stable:
+    uses:
```
Comment (stringhandler) [inline .github/workflows/ci.yml:32]: Not sure if this works, but we need this to run the jobs on self-hosted for better run times
```suggestion
    runs-on: [ self-hosted, ubuntu18.04-high-cpu ]
    uses:
```

Comment (stringhandler) [reply .github/workflows/ci.yml:32]: And same for other jobs

File: .github/workflows/reusable-tests.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      test-wasm:
+        # flag to run any wasm tests. Includes key manager
+        required: true
+        type: boolean
+
+
+jobs:
+  tests:
+    name: test
+    runs-on: ubuntu-18.04
```
Comment (stringhandler) [inline .github/workflows/reusable-tests.yml:20]: Or is it better to replace all of these?
```suggestion
    runs-on: [ self-hosted, ubuntu18.04-high-cpu ]
```

File: .github/workflows/ci.yml (hunk 1 above)
Comment (CjS77) [reply .github/workflows/ci.yml:32]: Probably, but I left everything as they're currently configured.

**Comment by CjS77** [comment]
Nuke this for now. GH needs to add additional flexibility to Actions before this is realistic with self-hosted runners

**Pull request closed without merging**